### PR TITLE
Model/16/blockchain class

### PR DESCRIPTION
**Pull request by archmagos**
• Basic Blockchain Class added.
• Create transactions method connects directly with the front-end.
• Transactions given unique hash.
• Sinatra controller updated to match new model.
• All tests pass with 100% coverage.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -2,13 +2,18 @@
...
 
 # Understands sending and receiving transaction hashes
 class BlockchainApp < Sinatra::Base
+  set :blockchain, Blockchain.new
+
   post '/blocks/create' do
     data = JSON.parse(request.body.read)['params']
-    Block.create(data['sender'], data['receiver'], data['value'])
+    settings.blockchain.create_transaction(data)
+    p settings.blockchain
```
Comment (jeff1108) [inline app.rb]: remove this debug check?

File: app.rb [hunk 2]
```diff
@@ -2,13 +2,18 @@
...
+    halt 200
```
Comment (jeff1108) [inline app.rb]: What this is?

File: lib/block.rb [hunk 3]
```diff
@@ -33,20 +33,20 @@ def self.create(sender, receiver, value)
     switch_db_if_test_env
     result = @con.exec 'INSERT INTO blocks (sender, receiver, value, hash, ' \
             "previous_tx) VALUES ('#{sender}', '#{receiver}', '#{value}', " \
-            "'#{Block.get_hash(sender, receiver, Block.prev_tx_hash)}', " \
```
Comment (jeff1108) [inline lib/block.rb]: Dan is working on stripping the database logic from the block class

File: spec/block_spec.rb [hunk 4]
```diff
@@ -18,8 +18,8 @@
   end
 
   describe '.previous_transaction' do
-    it 'returns the previous transaction hash' do
```
Comment (jeff1108) [inline spec/block_spec.rb]: overlap of work on block class

File: app.rb (hunk 1 above)
Comment (archmagos) [reply app.rb]: Good spot, will do

File: app.rb (hunk 2 above)
Comment (archmagos) [reply app.rb]: L14 tells the blockchain instance to create a new transaction with the post parameters
L15 I will remove
L16 Closes the post request with a server response of 200

File: lib/block.rb (hunk 3 above)
Comment (archmagos) [reply lib/block.rb]: Ok - I would just merge these for now and then Dan's branch can replace this

File: spec/block_spec.rb (hunk 4 above)
Comment (archmagos) [reply spec/block_spec.rb]: As above

Comment (archmagos) [reply spec/block_spec.rb]: As above. Just merge now, and replace with Dan's.

**Comment by archmagos** [comment]
Changes made. Take a look at my comments and can I get a re-review? Cheers

**Comment by jeff1108** [comment]
:shipit:

**Pull request merged**